script/create-test-appx.js

```diff
@@ -0,0 +1,224 @@
+
+/**
+ * Do we need to copy this file over?
+ *
+ * @param {String} name
+ * @returns {boolean}
+ */
+function isFileNeeded (name) {
+  return !/.*\.(?:(pdb)|(ilk)|(exp)|(lib))$/ig.test(name)
+}
+
+/**
+ * Do we need to copy this directory over?
+ *
+ * @param {String} name
+ * @returns {boolean}
+ */
+function isDirNeeded (name) {
+  return name === 'resources' || name === 'locales'
+}
+
+/**
+ * Starts a child process using the provided executable
+ *
+ * @param fileName      - Path to the executable to start
+ * @param args          - Arguments for spawn
+ * @param options       - Options passed to spawn
+ * @returns {Promise}   - A promise that resolves when the
+ *                      process exits
+ */
+function executeChildProcess (fileName, args, options) {
+  return new Promise((resolve, reject) => {
+    const child = childProcess.spawn(fileName, args, options)
+    const data = []
+
+    child.stdout.on('data', (line) => {
+      data.push(line)
+      console.log(line.toString())
+    })
+    child.stderr.on('data', (line) => {
+      data.push(line)
+      console.log(line.toString())
+    })
+
+    child.on('exit', (code) => {
+      if (code !== 0) {
+        return reject(new Error(fileName + ' exited with code: ' + code))
+      }
+      return resolve(data.join('\n'))
+    })
+
+    child.stdin.end()
+  })
+}
+
+/**
+ * Installs the test certificate to the local machine.
+ * Needs admin rights.
+ */
+async function installPfx () {
+  const installPfxArgs = [
+    'Import-PfxCertificate',
+    '-FilePath',
+    CERT_PATH,
+    '-CertStoreLocation',
+    '"Cert:\\LocalMachine\\TrustedPeople"'
+  ]
+
+  await executeChildProcess('powershell.exe', installPfxArgs)
+}
+
+/**
+ * Is the PFX certificate installed on the current machine?
+ *
+ * @returns {boolean}
+ */
+async function isPfxInstalled () {
+  const checkPfxArgs = [
+    '[bool](dir cert:\\LocalMachine\\TrustedPeople\\ | ? { $_.subject -like "cn=electrontest" })'
+  ]
+
+  const result = await executeChildProcess('powershell.exe', checkPfxArgs)
+
+  return result.toLowerCase().trim() === 'true'
+}
+
+/**
+ * Copy in our known shim files
+ *
+ * @returns {Promise<void>}
+ */
+async function copyInShim () {
+  const contents = await fs.readdir(SHIM_PATH)
+
+  for (const file of contents) {
+    const source = path.join(SHIM_PATH, file)
+    const desination = path.join(PREAPPX_PATH, file)
+
+    await fs.copy(source, desination)
+  }
+}
+
+/**
+ * Take the output directory and turn it into an appx
+ *
+ * @returns {Promise<void>}
+ */
+async function convert () {
+  fs.emptyDirSync(OUT_PATH)
+
+  // Install the test certificate
+  if (!(await isPfxInstalled())) {
+    console.warn(`The test certificate is not installed!`)
+    console.warn(`We will now try to install it. If this is not an administrative shell,`)
+    console.warn(`this operation will fail. If that's the case, please run this script`)
+    console.warn(`again as administrator`)
```

Comment:
_(minor)_ kinda weird having four different `console.warn` calls here. Could this be a single call with newlines in the string?